src/command_frame.cc

```diff
@@ -21,49 +36,96 @@ auto intelligence::jsonschema::cli::frame(
-    if (!entry.pointer.empty()) {
-      std::cout << " ";
-    }
+  const auto output_json = options.contains("json") || options.contains("j");
+  if (output_json) {
+    auto output_json_object = sourcemeta::jsontoolkit::JSON::make_object();
+    auto frame_json = sourcemeta::jsontoolkit::JSON::make_object();
+    auto references_json = sourcemeta::jsontoolkit::JSON::make_object();
 
-    sourcemeta::jsontoolkit::stringify(entry.pointer, std::cout);
-    std::cout << "\n";
-    std::cout << "    Base URI         : " << entry.base << "\n";
-    std::cout << "    Relative Pointer :";
-    if (!entry.relative_pointer.empty()) {
-      std::cout << " ";
+    for (const auto &[key, entry] : frame) {
+      auto frame_entry = sourcemeta::jsontoolkit::JSON::make_object();
+      frame_entry.assign("root",
+                         sourcemeta::jsontoolkit::JSON{entry.root.value_or(nullptr)});
+      std::ostringstream pointer_stream;
+      sourcemeta::jsontoolkit::stringify(entry.pointer, pointer_stream);
+      frame_entry.assign("pointer",
+                         sourcemeta::jsontoolkit::JSON{pointer_stream.str()});
+      frame_entry.assign("base", sourcemeta::jsontoolkit::JSON{entry.base});
+      frame_entry.assign("type", sourcemeta::jsontoolkit::JSON{enum_to_string(entry.type)});
+      std::ostringstream reference_stream;
+      sourcemeta::jsontoolkit::stringify(entry.relative_pointer, reference_stream);
+      frame_entry.assign("relativePointer",
+                         sourcemeta::jsontoolkit::JSON{reference_stream.str()});
+      frame_entry.assign("dialect", sourcemeta::jsontoolkit::JSON{entry.dialect});
+      frame_json.assign(key.second, sourcemeta::jsontoolkit::JSON{frame_entry});
     }
+    output_json_object.assign("frames", sourcemeta::jsontoolkit::JSON{frame_json});
 
-    sourcemeta::jsontoolkit::stringify(entry.relative_pointer, std::cout);
-    std::cout << "\n";
-    std::cout << "    Dialect          : " << entry.dialect << "\n";
-  }
-
-  for (const auto &[pointer, entry] : references) {
-    std::cout << "(REFERENCE) URI: ";
-    sourcemeta::jsontoolkit::stringify(pointer.second, std::cout);
-    std::cout << "\n";
-
-    std::cout << "    Type             : ";
-    if (pointer.first == sourcemeta::jsontoolkit::ReferenceType::Dynamic) {
-      std::cout << "Dynamic";
-    } else {
-      std::cout << "Static";
+    for (const auto &[pointer, entry] : references) {
+      auto ref_entry = sourcemeta::jsontoolkit::JSON::make_object();
+      ref_entry.assign("type", sourcemeta::jsontoolkit::JSON{
+                                   pointer.first ==
+                                           sourcemeta::jsontoolkit::ReferenceType::Dynamic
+                                       ? "Dynamic"
+                                       : "Static"});
+      ref_entry.assign("destination",
+                       sourcemeta::jsontoolkit::JSON{entry.destination});
+      if (entry.base.has_value()) {
```

Comment:
For every `has_value` check, can you add an `else` that sets it to null? Just to make it explicit that such property doesn't have a value